plugins/modules/cloud/docker/docker_stack_info.py

```diff
@@ -0,0 +1,122 @@
+
+# Copyright (c) 2018 Dario Zanzico (git@dariozanzico.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'status': ['preview'],
+    'supported_by': 'community',
+    'metadata_version': '1.1'
+}
+
+DOCUMENTATION = '''
+---
+module: docker_stack_info
+author: "Jose Angel Munoz (@imjoseangel)"
+short_description: docker stack module info
+description:
+  - Manage docker stacks using the 'docker stack' command
+    on the target node (see examples).
+version_added: "2.11"
+options:
+  name:
+    description:
+      - Stack name
+    type: str
+  action:
+    description:
+      - If set to C(tasks) displays the list of stacks using the C(docker stack ls) command.
+        If not set, the module use "list" by default.
+    type: str
+    choices: ["list", "tasks"]
+    default: list
+'''
+
+RETURN = '''
+msg:
+    description: |
+        dictionary containing the list of stacks or tasks associated
+        to a stack name.
+    sample: >
+        "msg": [{"name":"grafana","namespace":"default","orchestrator":"Kubernetes","services":"2"}]
+    returned: always
+    type: list
+'''
+
+EXAMPLES = '''
+  - name: Shows stack info
+    docker_stack_info:
+'''
+
+import ast
+from ansible.module_utils.basic import AnsibleModule
+from boto.dynamodb import item
+
+
+def docker_stack_list(module):
+    docker_bin = module.get_bin_path('docker', required=True)
+    rc, out, err = module.run_command([
+        docker_bin, "stack", "ls", "--format",
+        '{ "name": "{{.Name}}", "services": "{{.Services}}", "orchestrator": "{{.Orchestrator}}", "namespace": "{{.Namespace}}" }'
+    ])
+
+    return rc, out.strip(), err.strip()
+
+
+def docker_stack_tasks(module, name):
+    docker_bin = module.get_bin_path('docker', required=True)
+    rc, out, err = module.run_command([
+        docker_bin, "stack", "ps", name, "--format",
+        '{ "id": "{{.ID}}", "name": "{{.Name}}", "image": "{{.Image}}", "node": "{{.Node}}", "desired state": "{{.DesiredState}}", "current state": "{{.CurrentState}}", "error": "{{.Error}}", "ports": "{{.Ports}}" }'
+    ])
+    return rc, out.strip(), err.strip()
+
+
+def main():
+    module = AnsibleModule(argument_spec={
+        'name':
+        dict(type='str', required=False),
+        'action':
+        dict(type='str', choices=['list', 'tasks'], default="list"),
+    },
+                           supports_check_mode=False)
+
+    results = []
+    name = module.params['name']
+    action = module.params['action']
+
+    if action == 'tasks':
+        if not name:
+            module.fail_json(msg=("action requires a stack name"))
+
+        rc, out, err = docker_stack_tasks(module, name)
+
+    elif action == "list":
+        rc, out, err = docker_stack_list(module)
+
+    if rc != 0:
+        module.fail_json(msg="'docker stack down' command failed",
```

Comment:
This text seems wrong!

Reply:
The text is related to the stack. If the stack is down or not working properly, it will return the message.

Reply:
Change to out. I think it is better when returning different messages. Unfortunately the rc is the same whether the service is down or doesn't exist.